src/OpenFeature/OpenFeatureClient.cs

```diff
@@ -263,7 +263,23 @@ private async Task<FlagEvaluationDetails<T>> EvaluateFlagAsync<T>(
                     (await resolveValueDelegate.Invoke(flagKey, defaultValue, contextFromHooks.EvaluationContext, cancellationToken).ConfigureAwait(false))
                     .ToFlagEvaluationDetails();
 
-                await this.TriggerAfterHooksAsync(allHooksReversed, hookContext, evaluation, options, cancellationToken).ConfigureAwait(false);
+                if (evaluation.ErrorType == ErrorType.None)
+                {
+                    await this.TriggerAfterHooksAsync(
+                        allHooksReversed,
+                        hookContext,
+                        evaluation,
+                        options,
+                        cancellationToken
+                    ).ConfigureAwait(false);
+                }
+                else
+                {
+                    var exception = new FeatureProviderException(evaluation.ErrorType, evaluation.ErrorMessage);
+                    this.FlagEvaluationErrorWithDescription(flagKey, evaluation.ErrorType.GetDescription(), exception);
+                    await this.TriggerErrorHooksAsync(allHooksReversed, hookContext, exception, options, cancellationToken)
+                        .ConfigureAwait(false);

```

Comment:
Do we not need to assign the `evaluation` variable here? We do it when we get an exception, and at this stage, we don't do it when there is an error at the provider error.

Reply:
I'm not sure I understand your question. The `evaluation` variable is set in line 262, and the exception on this line produces no new information

Reply:
I meant that we override the `evaluation` variable with some "custom" values when there is an exception. In this case, we are just returning the result without overriding. Is this the expected behaviour?

If it is, I am happy to approve it. 👍

Reply:
That's a good question. I assumed that the initial value for evaluation will contain the proper data. Maybe someone else knows what to do here?

Reply:
Yes that is right, the information should be already contained in the evaluation details, as we do it in java and js